things/api.py

```diff
@@ -476,30 +485,46 @@ def today(**kwargs):
     """
     Read Today's tasks into dicts.
 
-    Note: This might not produce desired results if the Things app hasn't
-    been opened yet today and the yellow "OK" button clicked for new tasks.
-    In general, you can assume that whatever state the Things app was in
-    when you last opened it, that's the state reflected here by the API.
+    Note: This method is a bit tricky.
+    The Things database reflects the status of the Things app when it was last opened.
+    When you open the Things app, new to-dos might have appeared in the Today view.
+    Those to-dos are indicated by a yellow dot in the app.
+    We try to predict what those new to-dos would be and return them as well.
+    Some cases, however, are not yet covered. This includes repeating to-dos.
 
     See `things.api.tasks` for details on the optional parameters.
     """
-    database = pop_database(kwargs)
-    if not database.was_modified_today():  # pragma: no cover
-        print(
-            "[NOTE] The results reflect the state of the Things app "
-            "when it was last run. If the results seem out of date, "
-            "then run the Things app and click the yellow 'OK' button "
-            "to update Today's to-dos and projects.",
-            file=sys.stderr,
-        )
-    return tasks(
+    regular_today_tasks = tasks(
         start_date=True,
         start="Anytime",
         index="todayIndex",
-        database=database,
         **kwargs,
     )
 
+    unconfirmed_scheduled_tasks = tasks(
+        start_date="past",
+        start="Someday",
+        index="todayIndex",
+        **kwargs,
+    )
+
+    unconfirmed_overdue_tasks = tasks(
+        start_date=False,
+        deadline="past",
+        start="Anytime",
+        deadline_suppressed=False,
+        **kwargs,
+    )
+
+    result = [
+        *regular_today_tasks,
+        *unconfirmed_scheduled_tasks,
+        *unconfirmed_overdue_tasks,
+    ]
+    result.sort(key=lambda task: (task["today_index"], task["start_date"]))
```

Comment:
I don't see how the sorting here is correct. Aren't the yellow dots always on top?

Reply:
Just saw your earlier comment. All good. 👌